### PR TITLE
Uaw 58 get a booking

### PR DESCRIPTION
**Pull request by popsps**
Add Booking page
Add list all bookings page

### CODE REVIEW

File: frontend/admin-client/src/app/bookings/booking-list/booking-list.component.ts [hunk 1]
```diff
@@ -20,7 +20,14 @@ export class BookingListComponent implements OnInit {
     this.loading = true;
     this.bookingService.getAllBookings(environment.bookingApiUrl)
       .subscribe(bookings => {
-        console.log(bookings);
+        console.log('all bookings:', bookings);
+        bookings.forEach(booking => {
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking-list/booking-list.component.ts:24]: you could use .map here instead to make it clearer that you are mutating bookings/flights

**Review by Samnj583 — APPROVED** [review]

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 2]
```diff
@@ -30,20 +30,26 @@ export class BookingComponent implements OnInit {
     this.bookingService.getBookingById(environment.bookingApiUrl, this.bookingId)
       .subscribe(booking => {
         console.log(booking);
+        booking?.flights.forEach(flight => {
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:33]: same here

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 3]
```diff
@@ -30,20 +30,26 @@ export class BookingComponent implements OnInit {
...
+          flight.arrivalTime.setHours(Math.random() * 8 + 2 + flight.arrivalTime.getHours());
+        });
         this.booking = booking;
         this.initForm();
         this.loading = false;
       }, error => {
         this.loading = false;
-        this.error = {isError: true, message: error?.error?.message || error?.message, status: error?.status};
+        // this.error = {isError: true, message: error?.error?.message || error?.message, status: error?.status};
+        this.error = {isError: true, message: 'NO such a booking exists', status: 404};
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:44]: isError property seems redundant

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 4]
```diff
@@ -30,20 +30,26 @@ export class BookingComponent implements OnInit {
...
         console.log('error', error);
       });
 
   }
 
   submitUpdate(): void {
     console.log(JSON.stringify(this.booking));
-    this.booking.bookerId = this.booking.id;
+    // this.booking.bookerId = this.booking.id;
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:52]: probably shouldn't push commented changes

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 5]
```diff
@@ -53,7 +59,7 @@ export class BookingComponent implements OnInit {
   }
 
   deleteBooking(): void {
-    this.booking.bookerId = this.booking.id;
+    // this.booking.bookerId = this.booking.id;
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:62]: same here

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 6]
```diff
@@ -67,19 +73,20 @@ export class BookingComponent implements OnInit {
   }
 
   initForm(): void {
-    const {id, isActive, bookerId, confirmationCode, passengers} = this.booking;
+    // const {id, isActive, bookerId, confirmationCode, passengers} = this.booking;
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:76]: and here

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 7]
```diff
@@ -67,19 +73,20 @@ export class BookingComponent implements OnInit {
...
+    const {id, isActive, passengers, flights} = this.booking;
     this.bForm = this.fb.group({
       id,
       isActive,
-      confirmationCode,
-      bookerId,
+      // confirmationCode,
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:81]: same here

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 8]
```diff
@@ -67,19 +73,20 @@ export class BookingComponent implements OnInit {
...
+      // bookerId,
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:82]: same

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts [hunk 9]
```diff
@@ -67,19 +73,20 @@ export class BookingComponent implements OnInit {
...
       passengers: this.fb.array([]),
       flights: this.fb.array([])
     });
     passengers.forEach(passenger => {
       const passengerForm = this.fb.group({
         id: passenger?.id,
-        bookingId: passenger?.bookingId,
+        // bookingId: passenger?.bookingId,
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/bookings/booking/booking.component.ts:89]: double same

File: frontend/admin-client/src/app/shared/models/booking.ts [hunk 10]
```diff
@@ -1,16 +1,58 @@
...
+    given: string;
+    family: string;
+  };
+};
 
 export class Booking implements Deserializable {
   id: number;
   isActive: boolean;
-  confirmationCode: string;
+  // "USER" | "GUEST"
```
Comment (WesleyClements) [inline frontend/admin-client/src/app/shared/models/booking.ts:44]: same

**Review by WesleyClements — APPROVED** [review]

File: frontend/admin-client/src/app/bookings/booking/booking.component.ts (hunk 2 above)
Comment (popsps) [reply frontend/admin-client/src/app/bookings/booking/booking.component.ts:33]: Thanks.

**Pull request merged**